Loop/Managers/DeviceDataManager.swift

```diff
@@ -1291,100 +1291,110 @@ extension Notification.Name {
 extension DeviceDataManager {
     func handleRemoteNotification(_ notification: [String: AnyObject]) {
```

Comment:
I was trying to cleanup this function by removing the deep nesting, hence the additions of defer and a few new guards. 

The only real logical change is RemoteCommand now throws. That allows for improved logging when we fails to create a RemoteCommand, rather than just returning nil.

Reply:
Just some background on my usage of the throwing paradigm -- I've been adopting Swift Concurrency lately where throwing methods seem useful for simpler control flow down the asynchronous stack. i.e. Just `try` failable things in throwing methods and don't check for the return results until you must show an error to the user. It has been nice to not need to check/unwrap things, but only when you don't care why it fails, you just need to know it succeeded. That has been seeping into my non-concurrency world too like this PR

But I can see the rationale for using Result at times too (as I think is what you meant by returning the Error). I think the type guarantees you are suggesting are around using exhaustive switches on the exact error type to maybe take the next best action based on what kind of error happened.

I went ahead and updated this PR to use Result which returns the Error as I think you were suggesting, just to see how that looks for comparison. I did not end up leveraging it here yet as I am just logging the error as you mentioned but it is interesting to see the difference.